### PR TITLE
make tests simpler

### LINKED ISSUE #346: make tests simpler (reduce test data size) - facilitates faster builds for PRs
reduce checkerboard to only the edge case patches (make it super small - maybe 2 checkerboards across in all directions for train and test sets. 

modify checkerboard build job to run all batches and around 5 epochs

remove Dutch F3 test job entirely

### PR DESCRIPTION
**Pull request by fazamani**
This closes #346 

performance checks for patch_deconvnet is disabled (investigating a performance issue)

### CODE REVIEW

File: scripts/gen_checkerboard.py [hunk 1]
```diff
@@ -261,6 +289,8 @@ def main(args):
 parser.add_argument("--dataroot", help="Root location of the input data", type=str, required=True)
 parser.add_argument("--dataout", help="Root location of the output data", type=str, required=True)
 parser.add_argument("--box_size", help="Size of the bounding box", type=int, required=False, default=100)
+parser.add_argument("--based_on", help="This determines the shape of synthetic data array", type=str, required=False, default='dutch_f3')
+parser.add_argument("--box_number", help="Number of boxes", type=int, required=False, default=2)
```
Comment (maxkazmsft) [inline scripts/gen_checkerboard.py:293]: you can add a "choices" parameter like in "--type" below and drop the assert in the code

File: scripts/gen_checkerboard.py [hunk 2]
```diff
@@ -261,6 +289,8 @@ def main(args):
 parser.add_argument("--dataroot", help="Root location of the input data", type=str, required=True)
 parser.add_argument("--dataout", help="Root location of the output data", type=str, required=True)
 parser.add_argument("--box_size", help="Size of the bounding box", type=int, required=False, default=100)
+parser.add_argument("--based_on", help="This determines the shape of synthetic data array", type=str, required=False, default='dutch_f3')
```
Comment (maxkazmsft) [inline scripts/gen_checkerboard.py:292]: you can add a "choices" parameter like in "--type" below and drop the assert in the code

File: tests/cicd/main_build.yml [hunk 3]
```diff
@@ -169,7 +173,7 @@ jobs:
 
       # check validation set performance
       set -e
-      python ../../../../tests/cicd/src/check_performance.py --infile metrics_patch_deconvnet_no_depth.json
+      # python ../../../../tests/cicd/src/check_performance.py --infile metrics_patch_deconvnet_no_depth.json
```
Comment (maxkazmsft) [inline tests/cicd/main_build.yml:176]: we should enable this test - this is a base model

File: tests/cicd/main_build.yml [hunk 4]
```diff
@@ -258,7 +262,7 @@ jobs:
 
       # check test set performance
       set -e
-      python ../../../../tests/cicd/src/check_performance.py --infile metrics_test_patch_deconvnet_no_depth.json --test
+      # python ../../../../tests/cicd/src/check_performance.py --infile metrics_test_patch_deconvnet_no_depth.json --test
```
Comment (maxkazmsft) [inline tests/cicd/main_build.yml:265]: similar comment to enable

File: tests/cicd/main_build.yml (hunk 3 above)
Comment (fazamani) [reply tests/cicd/main_build.yml:176]: Thanks @maxkazmsft , I didn't mean to disable this permanently, this is the reproducibility problem that we are investigating! Will enable when we have a resolution!

File: scripts/gen_checkerboard.py (hunk 2 above)
Comment (fazamani) [reply scripts/gen_checkerboard.py:292]: @maxkazmsft  are you suggesting to change argument "based_on" to "choices"?

File: tests/cicd/main_build.yml [hunk 5]
```diff
@@ -267,162 +271,13 @@ jobs:
...
-      # Get return information for each pid
-      for file in "$dir"/*; do
-        printf 'PID %d returned %d\n' "${file##*/}" "$(<"$file")"
-        [[ "$(<"$file")" -ne "0" ]] && exit 1 || echo "pass"
-      done
-
-      # Remove the temporary directory
-      rm -r "$dir"
-
-      echo "PASSED"
```
Comment (yalaudah) [inline tests/cicd/main_build.yml:418]: Sorry I might be missing something obvious, but why are thes tests removed?

File: tests/cicd/src/scripts/get_data_for_builds.sh [hunk 6]
```diff
@@ -50,4 +50,4 @@ python prepare_dutchf3.py split_train_val patch   --data_dir=${DATA_F3} --label_
 DATA_CHECKERBOARD="${DATA_CHECKERBOARD}/data"
 # repeat for checkerboard dataset
 python prepare_dutchf3.py split_train_val section --data_dir=${DATA_CHECKERBOARD} --label_file=train/train_labels.npy --output_dir=splits --split_direction=both
-python prepare_dutchf3.py split_train_val patch   --data_dir=${DATA_CHECKERBOARD} --label_file=train/train_labels.npy --output_dir=splits --stride=50 --patch_size=100 --split_direction=both
+python prepare_dutchf3.py split_train_val patch   --data_dir=${DATA_CHECKERBOARD} --label_file=train/train_labels.npy --output_dir=splits --stride=50 --patch_size=100 --split_direction=both --section_stride=100
```
Comment (yalaudah) [inline tests/cicd/src/scripts/get_data_for_builds.sh:53]: The section stride is very big. This might lead to a really small training set.. it will make training faster, but if we are testing the performance of this model, it might note perform well at all.

File: tests/cicd/main_build.yml (hunk 4 above)
Comment (fazamani) [reply tests/cicd/main_build.yml:267]: as discussed, I added a "TODO" to enable this after investigation!

File: tests/cicd/main_build.yml (hunk 3 above)
Comment (fazamani) [reply tests/cicd/main_build.yml:177]: same, we will enable this after investigation!

File: tests/cicd/src/scripts/get_data_for_builds.sh (hunk 6 above)
Comment (fazamani) [reply tests/cicd/src/scripts/get_data_for_builds.sh:53]: @yalaudah I mark this as resolved per our chat :)

File: tests/cicd/main_build.yml (hunk 5 above)
Comment (fazamani) [reply tests/cicd/main_build.yml:443]: @yalaudah I mark this as resolved per our chat :)
